plugins/inputs/nftables/nftables.go

```diff
@@ -0,0 +1,185 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package nftables
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const measurement = "nftables"
+
+type Nftables struct {
+	UseSudo bool   `toml:"use_sudo"`
+	Binary  string `toml:"binary"`
+	Tables  []string
+}
+
+var nftableConfig = `
+  ## Configuration for nftables
+  tables = [ "filter" ]
+`
+
+type Nftable struct {
+	Metainfo          *Metainfo
+	Rules             []*Rule
+	JSONSchemaVersion int `json:"json_schema_version"`
+}
+
+// UnmarshalJSON handles custom parsing of the nftable output which is
+// designed in a generic that is not compatible with the generic parser.
+func (nftable *Nftable) UnmarshalJSON(b []byte) error {
+	var atable map[string][]map[string]json.RawMessage
+	if err := json.Unmarshal(b, &atable); err != nil {
+		return fmt.Errorf("unable to unmarshal: %s", b)
+	}
+	// []map[string]interface
+	nfthings := atable["nftables"]
+	for _, nfthing := range nfthings {
+		hasKey := func(key string) bool { _, ok := nfthing[key]; return ok }
+		switch {
+		case hasKey("metainfo"):
+			var mi Metainfo
+			err := json.Unmarshal(nfthing["metainfo"], &mi)
+			if err != nil {
+				return fmt.Errorf("unable to parse metadata: %w", err)
+			}
+			nftable.Metainfo = &mi
+		case hasKey("rule"):
+			var r Rule
+			err := json.Unmarshal(nfthing["rule"], &r)
+			if err != nil {
+				return fmt.Errorf("unable to parse rule: %w", err)
+			}
+			nftable.Rules = append(nftable.Rules, &r)
+		default:
+			// something we are not parsing
+		}
+	}
+	return nil
+}
+
+type Metainfo struct {
+	Version           string `json:"version"`
+	ReleaseName       string `json:"release_name"`
+	JSONSchemaVersion int    `json:"json_schema_version"`
+}
+
+type Rule struct {
```

Comment:
Missing JSON tags on Rule struct, Should have JSON tags for consistency

Reply:
I don't believe Rule should have JSON tags since they would not be utilized since we have a custom `UnmarshalJSON` func. I found that the redundant tags made the code more confusing. Thoughts?